### PR TITLE
add flag serial command

### PR DESCRIPTION
**Pull request by GitMeGet**
- updated TS_Settings to include upload_flag

### CODE REVIEW

File: alpha/serial_cmd.cpp [hunk 1]
```diff
@@ -209,6 +217,72 @@ static int do_clock_cmd(int argc, char **argv)
...
+      if (pass_strlen <= STR_ARG_MAXLEN)
+      {
+        strcpy(settings->wifiPass, wifi_args.password->sval[0]);
+      }
+      else
+      {
+        printf("Password exceeds max string length of %d\n\n", STR_ARG_MAXLEN);
+      }
+    }
+    TS_Storage.settings_save();
```
Comment (jamesi8086) [inline alpha/serial_cmd.cpp]: Oh wait, can you make save a separate command?

We need to be able to change settings w/o saving to flash. This is particularly useful during automated testing.

File: alpha/serial_cmd.cpp [hunk 2]
```diff
@@ -226,6 +300,95 @@ static void register_clock_cmd()
...
+        break;
+      case 1:
+        settings->upload_flag = true;
+        break;
+      default:
+        print_cmd_help(argv[0], (void**) &flag_args);
+        return ESP_ERR_INVALID_ARG;
+    }
+
+    TS_Storage.settings_save();
```
Comment (jamesi8086) [inline alpha/serial_cmd.cpp]: same

**Review by jamesi8086 — APPROVED** [review]
note on the save command

File: alpha/serial_cmd.cpp (hunk 1 above)
Comment (GitMeGet) [reply alpha/serial_cmd.cpp]: Got it. So perhaps default cmd saves to EEPROM and --ram saves to RAM

**Comment by GitMeGet** [comment]
added RAM flag

File: alpha/serial_cmd.cpp (hunk 1 above)
Comment (yongkimleng) [reply alpha/serial_cmd.cpp]: ah no, just have an explicit command that saves

we actually have a settings_load to overwrite whatever is in RAM w/ whats in flash

Comment (jamesi8086) [reply alpha/serial_cmd.cpp]: your method works

File: alpha/serial_cmd.cpp [hunk 3]
```diff
@@ -450,8 +451,7 @@ static int do_userid_cmd(int argc, char **argv)
   else if (userid_args.set->count == 1)
   {
     check_copy_str_setting(userid_args.set->sval[0], settings->userId);
-    TS_Storage.settings_save();
-    printf("Setting saved\n\n");
+    save_to_RAM_or_EEPROM(userid_args.ram_flag);
```
Comment (jamesi8086) [inline alpha/serial_cmd.cpp]: no mixed cases in function name

File: alpha/serial_cmd.cpp [hunk 4]
```diff
@@ -241,8 +241,7 @@ static int do_clock_cmd(int argc, char **argv)
   } 
   else
   {
-    printf("Type help for help\n\n");
-    return ESP_ERR_INVALID_ARG;
+    print_cmd_help(argv[0], (void**) &clock_args);
```
Comment (jamesi8086) [inline alpha/serial_cmd.cpp]: clockArgs

**Comment by jamesi8086** [comment]
I probably should have written a style guide:

filenames: snake_case_filenames
function names: snake_case
variable names: camelCase
type names: PascalCase with the exception of adding prefixes, eg. TS_ or _
defines: UPPER_CASE

**Comment by GitMeGet** [comment]
If using platformio, reduce the debug_level in platformio.ini to make it easier for your eyes :)

**Comment by jamesi8086** [comment]
can merge if no conflicts

**Pull request merged**